src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ExceptionTest.cs

```diff
@@ -131,11 +131,25 @@ public static void WarningsBeforeRowsTest()
         private static bool CheckThatExceptionsAreDistinctButHaveSameData(SqlException e1, SqlException e2)
         {
             Assert.True(e1 != e2, "FAILED: verification of exception cloning in subsequent connection attempts");
-
             Assert.False((e1 == null) || (e2 == null), "FAILED: One of exceptions is null, another is not");
 
             bool equal = (e1.Message == e2.Message) && (e1.HelpLink == e2.HelpLink) && (e1.InnerException == e2.InnerException)
-                && (e1.Source == e2.Source) && (e1.Data.Count == e2.Data.Count) && (e1.Errors == e2.Errors);
+                && (e1.Source == e2.Source) && (e1.Data.Count == e2.Data.Count) && (e1.Errors.Count == e2.Errors.Count);
+
+            for (int i = 0; i < e1.Errors.Count; i++)
+            {
+                equal = e1.Errors[i].Number == e2.Errors[i].Number
+                    && e1.Errors[i].Message == e2.Errors[i].Message
```

Comment:
Do we need the null checker here and also in other values in the loop?

Reply:
Not really since e1/e2 always contain "Errors" collection with at-least 1 error which is the actual exception.